tests/components/unifiprotect/test_camera.py

```diff
@@ -0,0 +1,358 @@
+        hass, camera_high_only, 0, secure=False, rtsp_enabled=True, enabled=False
+    )
+
+    await validate_camera_entity(
+        hass, camera_all_channels, 0, secure=True, rtsp_enabled=True, enabled=True
+    )
+    await validate_camera_entity(
+        hass, camera_all_channels, 0, secure=False, rtsp_enabled=True, enabled=False
+    )
+    await validate_camera_entity(
+        hass, camera_all_channels, 1, secure=True, rtsp_enabled=True, enabled=False
+    )
+    await validate_camera_entity(
+        hass, camera_all_channels, 1, secure=False, rtsp_enabled=True, enabled=False
+    )
+    await validate_camera_entity(
+        hass, camera_all_channels, 2, secure=True, rtsp_enabled=True, enabled=False
+    )
+    await validate_camera_entity(
+        hass, camera_all_channels, 2, secure=False, rtsp_enabled=True, enabled=False
+    )
+
+    await validate_camera_entity(
+        hass, camera_no_channels, 0, secure=True, rtsp_enabled=False, enabled=True
+    )
+
+
+async def test_missing_channels(
+    hass: HomeAssistant, mock_entry: MockEntityFixture, mock_camera: ProtectCamera
+):
+    """Test setting up camera with no camera channels."""
+
+    camera = mock_camera.copy(deep=True)
+    camera.channels = []
+
+    mock_entry.api.bootstrap.cameras = {camera.id: camera}
+
+    await hass.config_entries.async_setup(mock_entry.entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_registry = er.async_get(hass)
+
+    assert len(hass.states.async_all()) == 0
+    assert len(entity_registry.entities) == 0
+
+
+async def test_camera_image(
+    hass: HomeAssistant,
+    mock_entry: MockEntityFixture,
+    simple_camera: tuple[Camera, str],
+):
+    """Test retrieving camera image."""
+
+    mock_entry.api.get_camera_snapshot = AsyncMock()
+
+    await async_get_image(hass, simple_camera[1])
+    mock_entry.api.get_camera_snapshot.assert_called_once()
+
+
+async def test_camera_generic_update(
+    hass: HomeAssistant,
+    mock_entry: MockEntityFixture,
+    simple_camera: tuple[ProtectCamera, str],
+):
+    """Tests generic entity update service."""
+
+    assert await async_setup_component(hass, "homeassistant", {})
+
+    data: ProtectData = hass.data[DOMAIN][mock_entry.entry.entry_id]
+    assert data
+    assert data.last_update_success
+
+    state = hass.states.get(simple_camera[1])
+    assert state and state.state == "idle"
+
+    mock_entry.api.update = AsyncMock(return_value=None)
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
+        {ATTR_ENTITY_ID: simple_camera[1]},
+        blocking=True,
+    )
+
+    state = hass.states.get(simple_camera[1])
+    assert state and state.state == "idle"
+
+
+async def test_camera_interval_update(
+    hass: HomeAssistant,
+    mock_entry: MockEntityFixture,
+    simple_camera: tuple[ProtectCamera, str],
+):
+    """Interval updates updates camera entity."""
+
+    data: ProtectData = hass.data[DOMAIN][mock_entry.entry.entry_id]
```

Comment:
Please don't access integration details like hass.data in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations